src/main/java/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,58 @@
+package bookstore.controller;
+
+import bookstore.dto.BookDtoWithoutCategoryIds;
+import bookstore.dto.category.CategoryDto;
+import bookstore.dto.category.CreateCategoryRequestDto;
+import bookstore.service.book.BookService;
+import bookstore.service.category.CategoryService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("api/category")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping
+    public CategoryDto createCategory(CreateCategoryRequestDto requestDto) {
+        return categoryService.save(requestDto);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @PostMapping
+    public List<CategoryDto> getAll(Pageable pageable) {
+        return categoryService.findAll(pageable);
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @PostMapping
+    public CategoryDto getCategoryById(Long id) {
+        return categoryService.getById(id);
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping
```

Comment:
Please pay attetion to request type, when you create methods

Reply:
not fixed